sync/src/tests/private_transactions.rs

```diff
@@ -1,150 +0,0 @@
-use ethcore::client::{BlockChainClient, BlockId, ClientIoMessage};
-use ethcore::spec::Spec;
-use ethcore::miner::MinerService;
-use transaction::{Transaction, Action};
-use ethcore::test_helpers::push_block_with_transactions;
-use private_transactions::{ProviderConfig};
-use ethcore::account_provider::AccountProvider;
-use ethkey::KeyPair;
-use tests::helpers::*;
-use ethereum_types::{U256, Address};
-use rustc_hex::FromHex;
-use SyncConfig;
-
-fn contract_address(sender: &Address, nonce: &U256) -> Address {
-	let mut stream = RlpStream::new_list(2);
-	stream.append(sender);
-	stream.append(nonce);
-	From::from(keccak(stream.as_raw()))
-}
-
-fn seal_spec() -> Spec {
-	let spec_data = r#"
-	{
-		"name": "PrivateTransactions",
-		"engine": {
-			"instantSeal": null
-		},
-		"params": {
-			"gasLimitBoundDivisor": "0x0400",
-			"accountStartNonce": "0x0",
-			"maximumExtraDataSize": "0x20",
-			"minGasLimit": "0x1388",
-			"networkID" : "0x11"
-		},
-		"genesis": {
-			"seal": {
-				"generic": "0x0"
-			},
-			"difficulty": "0x20000",
-			"author": "0x0000000000000000000000000000000000000000",
-			"timestamp": "0x00",
-			"parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
-			"extraData": "0x",
-			"gasLimit": "0x989680"
-		},
-		"accounts": {
-			"0000000000000000000000000000000000000001": { "balance": "1", "builtin": { "name": "ecrecover", "pricing": { "linear": { "base": 3000, "word": 0 } } } },
-			"0000000000000000000000000000000000000002": { "balance": "1", "builtin": { "name": "sha256", "pricing": { "linear": { "base": 60, "word": 12 } } } },
-			"0000000000000000000000000000000000000003": { "balance": "1", "builtin": { "name": "ripemd160", "pricing": { "linear": { "base": 600, "word": 120 } } } },
-			"0000000000000000000000000000000000000004": { "balance": "1", "builtin": { "name": "identity", "pricing": { "linear": { "base": 15, "word": 3 } } } }
-		}
-	}
-	"#;
-
-	Spec::load(&::std::env::temp_dir(), spec_data.as_bytes()).unwrap()
-}
-
-#[test]
-fn send_private_transaction() {
-	// Setup two clients
-	let s0 = KeyPair::from_secret_slice(&keccak("1")).unwrap();
-	let s1 = KeyPair::from_secret_slice(&keccak("0")).unwrap();
-	let ap = Arc::new(AccountProvider::transient_provider());
-	ap.insert_account(s0.secret().clone(), "").unwrap();
-	ap.insert_account(s1.secret().clone(), "").unwrap();
-
-	// Setup providers
-	let validator_config = ProviderConfig{
-		validator_accounts: vec![s1.address()],
-		signer_account: None,
-		passwords: vec!["".into()],
-	};
-
-	let signer_config = ProviderConfig{
-		validator_accounts: Vec::new(),
-		signer_account: Some(s0.address()),
-		passwords: vec!["".into()],
-	};
-
-	let mut net = TestNet::with_spec_and_accounts(2, SyncConfig::default(), seal_spec, Some(ap.clone()), Some(vec![signer_config, validator_config]));
-	let client0 = net.peer(0).chain.clone();
-	let client1 = net.peer(1).chain.clone();
-	let provider0 = net.peer(0).private_provider.clone().unwrap().clone();
```

Comment:
imo, this is a bad tests. It does not prove that private_transactions are working. it's partly caused by the fact how our api looks like currently

This test mutates several objects (client, miner, engine, net) and assumes that real code will mutate these objects in the same way before executing private transaction

Reply:
It's not the test for private transactions as a whole functionality. It's mostly for its sync part (that's why it's placed here) to test, that messages communicated properly between clients.

Reply:
IMHO even partial test is better than none, so removing it without providing alternative is not so good.